src/Core/Auth/Repositories/Cosmos/GrantRepository.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System.Net;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Bit.Core.Auth.Models.Data;
+using Bit.Core.Settings;
+using Bit.Core.Utilities;
+using Microsoft.Azure.Cosmos;
+
+namespace Bit.Core.Auth.Repositories.Cosmos;
+
+public class GrantRepository : IGrantRepository
+{
+    private readonly CosmosClient _client;
+    private readonly Database _database;
+    private readonly Container _container;
+
+    public GrantRepository(GlobalSettings globalSettings)
+        : this(globalSettings.IdentityServer.CosmosConnectionString)
+    { }
+
+    public GrantRepository(string cosmosConnectionString)
+    {
+        var options = new CosmosClientOptions
+        {
+            Serializer = new SystemTextJsonCosmosSerializer(new JsonSerializerOptions
+            {
+                DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull,
+                WriteIndented = false
+            })
+        };
+        _client = new CosmosClient(cosmosConnectionString, options);
```

Comment:
The Cosmos grant repository is a dependency of the `PersistedGrantStore`, which is registered as a singleton.

We can't register the cosmos grant repository to DI as IGrantRepository since we need multiple instances of `IGrantRepository`. Until we move to .NET 8 with keyed services, I don't think there is a better solution that what we're doing right now. Same with CosmosClient. I think we need to be able to key that. I can add some comments.